### PR TITLE
Fix version actions

### PR DESCRIPTION
**Pull request by cdce8p**
Read version from `setup.cfg` if no `setup.py` file exists.

Required for
https://github.com/home-assistant/core/pull/66023
https://github.com/home-assistant/frontend/pull/11593

/CC: @ludeeus

### CODE REVIEW

File: helpers/version/action.yml [hunk 1]
```diff
@@ -51,7 +51,12 @@ runs:
           python3 -m pip install packaging
           python3 -m pip install .
           python3 script/version_bump.py nightly
-          version="$(python setup.py -V)"
+
+          if [[ -e setup.py ]]; then
+            version="$(python3 setup.py -V)"
+          else
+            version="$(python3 ${{ github.action_path }}/../read_version_cfg.py setup.cfg)"
+          fi;
```
Comment (ludeeus) [inline helpers/version/action.yml:59]: This only runs on core dev which has the `setup.cfg` file already, why have this logic and not check it directly?

File: helpers/read_version_cfg.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from pathlib import Path
+
+
+def main(argv: list[str] | None = None) -> int:
+    """Read version from setup.cfg file passed as first argument."""
+    argv = argv or sys.argv[1:]
+
+    parser = configparser.ConfigParser()
+    if (len(argv) < 1 or not Path(argv[0]).is_file()):
+        return 1
```
Comment (ludeeus) [inline helpers/read_version_cfg.py:14]:
```suggestion
    if (len(argv) < 1 or not Path(argv[0]).is_file()):
        return 1
    parser = configparser.ConfigParser()
```

File: helpers/read_version_cfg.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+from __future__ import annotations
+
+import configparser
+import sys
+from pathlib import Path
+
+
+def main(argv: list[str] | None = None) -> int:
+    """Read version from setup.cfg file passed as first argument."""
+    argv = argv or sys.argv[1:]
```
Comment (ludeeus) [inline helpers/read_version_cfg.py:10]: `argv` does not exist before this point, why check it?

Comment (cdce8p) [reply helpers/read_version_cfg.py:10]: That's the default pattern I tend to use for scripts. If the method is at some point used directly, it's often useful to pass an argument list as first argument.

File: helpers/version/action.yml (hunk 1 above)
Comment (cdce8p) [reply helpers/version/action.yml:59]: > This only runs on core dev [...]

I wasn't sure about that. A quick Github search showed that 61 repos use that action, including multiple inside the Home Assistant org. Not all might be updated yet. I think this is an elegant way to implement it backwards compatible.

https://github.com/home-assistant/supervisor/blob/main/setup.py
https://github.com/home-assistant/supervisor/blob/main/.github/workflows/builder.yml#L60-L64

Comment (ludeeus) [reply helpers/version/action.yml:59]:
```
elif [[ "${version}" == "dev" && "${{ inputs.type }}" == "core" ]]; then
```
Is what takes you to this section.

Comment (cdce8p) [reply helpers/version/action.yml]: Updated!

**Comment by cdce8p** [comment]
@ludeeus Would you mind taking a final look at this? I think it's ready to go.

File: helpers/read_version_cfg.py (hunk 3 above)
Comment (ludeeus) [reply helpers/read_version_cfg.py:10]: Then we should adjust that if that ever will be the case.

**Review by ludeeus — APPROVED** [review]

**Pull request merged**